src/core/server/i18n/i18n_service.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { take } from 'rxjs/operators';
+import { Logger } from '../logging';
+import { IConfigService } from '../config';
+import { CoreContext } from '../core_context';
+import { config as i18nConfigDef, I18nConfigType } from './i18n_config';
+import { getKibanaTranslationFiles } from './get_kibana_translation_files';
+import { initTranslations } from './init_translations';
+
+interface SetupDeps {
+  pluginPaths: string[];
+}
+
+/**
+ * @public
+ */
+export interface I18nServiceSetup {
+  /**
+   * Return the locale currently in use.
+   */
+  getLocale(): string;
+
+  /**
+   * Return the list of translation files currently in use.
+   */
+  getTranslationFiles(): string[];
```

Comment:
I need that to be accessible for the `localization` usage collector.

The `getLocale` is not really necessary as the collector could use `i18n.getLocale()` instead, but from an architectural point of view, I think it is better to have core be the entry point for that (and the contract is needed anyway for `getTranslationFiles` so it did not cost much to also add an additional API to it)